Bottom: d64bb997ee449d5e3031f4084e0ec0044a05f6ce
Top:    639a2d5d1e52f0b1accbc082595f1def1f9b16ba
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-10 12:19:57 -0500

Use docker named networks


---
diff --git a/ansible-playbook/group_vars/docker-hosts/docker.yml b/ansible-playbook/group_vars/docker-hosts/docker.yml
index 9acd909a..c2ac4ebd 100644
--- a/ansible-playbook/group_vars/docker-hosts/docker.yml
+++ b/ansible-playbook/group_vars/docker-hosts/docker.yml
@@ -15,3 +15,6 @@ docker_restore_config_base_dir: '/etc/docker_container.conf/{{ docker_image_tag
 
 # restore directories to temporarly store data being restored into docker containers
 docker_backup_dir: '/tmp/docker'
+
+# Network names
+docker_network_frontend: 'frontend'
diff --git a/ansible-playbook/site.yml b/ansible-playbook/site.yml
index 4df32b39..2ec6350c 100644
--- a/ansible-playbook/site.yml
+++ b/ansible-playbook/site.yml
@@ -29,6 +29,10 @@
     role: docker-build-base-image
     tags:
     - base_images
+  - name: Configure Docker networking
+    role: docker-networking
+    tags:
+    - docker_networking
   - name: Build and start the SVN Docker container
     role: docker-svn
     svn_repos: 'ddio NCD_Release'
diff --git a/roles/docker-exim4/defaults/main.yml b/roles/docker-exim4/defaults/main.yml
index 8761e475..7262e16e 100644
--- a/roles/docker-exim4/defaults/main.yml
+++ b/roles/docker-exim4/defaults/main.yml
@@ -50,3 +50,6 @@ email_accounts:
   mantisbt:
     user: mantis
     password: '{{ mantisbt_email_password }}'
+
+# Network names
+docker_network_frontend: 'frontend'
diff --git a/roles/docker-exim4/tasks/main.yml b/roles/docker-exim4/tasks/main.yml
index 379fdb6a..f00fdf9f 100644
--- a/roles/docker-exim4/tasks/main.yml
+++ b/roles/docker-exim4/tasks/main.yml
@@ -106,6 +106,8 @@
   docker_container:
     name: '{{ exim4_container_name }}'
     hostname: '{{ container_addr_map.exim4.hostname }}'
+    networks:
+      - name: '{{ docker_network_frontend }}'
     ports: '{{ container_port_map.exim4.port_args }}'
     image: '{{ exim4_image_name }}:{{ docker_image_tag }}'
     restart_policy: always
diff --git a/roles/docker-gitlab/defaults/main.yml b/roles/docker-gitlab/defaults/main.yml
index 3a0aec35..b742c18f 100644
--- a/roles/docker-gitlab/defaults/main.yml
+++ b/roles/docker-gitlab/defaults/main.yml
@@ -45,3 +45,6 @@ gitlab_db_dv_name: '{{ docker_name_prefix }}gitlab_db_DV'
 # restore directories to temporarly store data being restored into docker containers
 gitlab_docker_backup_dir: '{{ docker_backup_dir }}/GIT'
 host_gitlab_docker_restore_dir: '{{ bacula_dest }}{{ gitlab_docker_backup_dir }}'
+
+# Network names
+docker_network_frontend: 'frontend'
diff --git a/roles/docker-gitlab/tasks/main.yml b/roles/docker-gitlab/tasks/main.yml
index 0e7407ba..a91a3a6a 100644
--- a/roles/docker-gitlab/tasks/main.yml
+++ b/roles/docker-gitlab/tasks/main.yml
@@ -130,6 +130,8 @@
     detach: true
     restart_policy: always
     image: '{{ redis_image_name }}:{{ docker_image_tag }}'
+    networks:
+      - name: '{{ docker_network_frontend }}'
 
 - name: start container (postgres)
   docker_container:
@@ -138,6 +140,8 @@
     restart_policy: always
     volumes_from: '{{ gitlab_db_dv_name }}'
     image: '{{ gitlab_db_image_name }}:{{ docker_image_tag }}'
+    networks:
+      - name: '{{ docker_network_frontend }}'
 
 - name: start container (gitlab)
   docker_container:
@@ -146,11 +150,8 @@
     restart_policy: always
     hostname: '{{ container_addr_map.git.hostname }}'
     ports: '{{ container_port_map.git.port_args }}'
-    link:
-      - '{{ openldap_container_name }}:gitlab-ldap'
-      - '{{ gitlab_redis_container_name }}:gitlab-redis'
-      - '{{ gitlab_db_container_name }}:gitlab-db'
-      - '{{ exim4_container_name }}:{{ container_addr_map.exim4.hostname }}'
+    networks:
+      - name: '{{ docker_network_frontend }}'
     volumes-from:
       - '{{ openssl_dv_name }}'
       - '{{ gitlab_dv_name }}'
diff --git a/roles/docker-mantisbt/defaults/main.yml b/roles/docker-mantisbt/defaults/main.yml
index 896fa010..e4df7a7f 100644
--- a/roles/docker-mantisbt/defaults/main.yml
+++ b/roles/docker-mantisbt/defaults/main.yml
@@ -47,3 +47,6 @@ host_mantisbt_docker_restore_dir: '{{ bacula_dest }}{{ mantisbt_docker_backup_di
 mantisbt_static_backup_file: '/mantisbt.tar'
 mantisbt_database_backup_file: '/mantisbtdb.sql'
 mantisbt_database_files_table_backup_file: '/mantisbtdb-files_table.sql'
+
+# Network names
+docker_network_frontend: 'frontend'
diff --git a/roles/docker-mantisbt/tasks/main.yml b/roles/docker-mantisbt/tasks/main.yml
index 3352d62c..7d183bc1 100644
--- a/roles/docker-mantisbt/tasks/main.yml
+++ b/roles/docker-mantisbt/tasks/main.yml
@@ -148,6 +148,8 @@
     volumes_from:
       - '{{ mantisbt_db_dv_name }}'
     image: '{{ mantisbt_db_image_name }}:{{ docker_image_tag }}'
+    networks:
+      - name: '{{ docker_network_frontend }}'
 
 - name: start container (mantisbt)
   docker_container:
@@ -156,10 +158,8 @@
     restart_policy: always
     hostname: '{{ container_addr_map.mantisbt.hostname }}'
     ports: '{{ container_port_map.mantisbt.port_args }}'
-    link: 
-      -'{{ openldap_container_name }}:mantisbt_ldap'
-      -'{{ mantisbt_db_container_name }}:mantisbt_db'
-      -'{{ exim4_container_name }}:{{ container_addr_map.exim4.hostname }}'
+    networks:
+      - name: '{{ docker_network_frontend }}'
     volumes-from: 
       - '{{ openssl_dv_name }}'
       - '{{ mantisbt_dv_name }}'
diff --git a/roles/docker-mediawiki/defaults/main.yml b/roles/docker-mediawiki/defaults/main.yml
index 2a5319d5..dbfe9716 100644
--- a/roles/docker-mediawiki/defaults/main.yml
+++ b/roles/docker-mediawiki/defaults/main.yml
@@ -45,3 +45,6 @@ host_wiki_docker_restore_dir: '{{ bacula_dest }}{{ wiki_docker_backup_dir }}'
 # files restored from tape
 wiki_static_backup_file: '/mediawiki.tar'
 wiki_database_backup_file: '/wikidb.sql'
+
+# Network names
+docker_network_frontend: 'frontend'
diff --git a/roles/docker-mediawiki/tasks/main.yml b/roles/docker-mediawiki/tasks/main.yml
index 87463776..60286ff5 100644
--- a/roles/docker-mediawiki/tasks/main.yml
+++ b/roles/docker-mediawiki/tasks/main.yml
@@ -121,6 +121,8 @@
     restart_policy: always
     volumes_from: '{{ wiki_db_dv_name }}'
     image: '{{ wiki_db_image_name }}:{{ docker_image_tag }}'
+    networks:
+      - name: '{{ docker_network_frontend }}'
 
 - name: start container (mediwiki)
   docker_container:
@@ -129,10 +131,8 @@
     restart_policy: always
     hostname: '{{ container_addr_map.wiki.hostname }}' 
     ports: '{{ container_port_map.wiki.port_args }}'
-    link:
-      - '{{ openldap_container_name }}:wiki_ldap'
-      - '{{ wiki_db_container_name }}:wiki_db'
-      - '{{ exim4_container_name }}:{{ container_addr_map.exim4.hostname }}'
+    networks:
+      - name: '{{ docker_network_frontend }}'
     volumes-from: 
       - '{{ openssl_dv_name }}'
       - '{{ wiki_dv_name }}'
diff --git a/roles/docker-networking/defaults/main.yml b/roles/docker-networking/defaults/main.yml
new file mode 100644
index 00000000..f480011b
--- /dev/null
+++ b/roles/docker-networking/defaults/main.yml
@@ -0,0 +1,5 @@
+---
+# defaults file for docker-networking
+
+# Network names
+docker_network_frontend: 'frontend'
diff --git a/roles/docker-networking/tasks/main.yml b/roles/docker-networking/tasks/main.yml
new file mode 100644
index 00000000..595e8bb7
--- /dev/null
+++ b/roles/docker-networking/tasks/main.yml
@@ -0,0 +1,7 @@
+---
+# tasks file for docker-networking
+
+- name: Create frontend network for services
+  docker_network:
+    name: '{{ docker_network_frontend }}'
+    state: present
diff --git a/roles/docker-openldap/defaults/main.yml b/roles/docker-openldap/defaults/main.yml
index 624a6e9d..5f41fba7 100644
--- a/roles/docker-openldap/defaults/main.yml
+++ b/roles/docker-openldap/defaults/main.yml
@@ -51,3 +51,6 @@ phpldapadmin_image_name: 'osixia/phpldapadmin:latest'
 
 # the name of the container being started
 phpldapadmin_container_name: '{{ docker_name_prefix }}phpldapadmin'
+
+# Network names
+docker_network_frontend: 'frontend'
diff --git a/roles/docker-openldap/tasks/main.yml b/roles/docker-openldap/tasks/main.yml
index 2d0c7232..32b4a575 100644
--- a/roles/docker-openldap/tasks/main.yml
+++ b/roles/docker-openldap/tasks/main.yml
@@ -120,6 +120,8 @@
     env:
       LDAP_HOSTNAME: '{{ container_addr_map.ldap.hostname }}'
     image: '{{ openldap_image_name }}:{{ docker_image_tag }}'
+    networks:
+      - name: '{{ docker_network_frontend }}'
 
 - name: start container (phpldapadmin)
   docker_container:
@@ -134,7 +136,8 @@
       PHPLDAPADMIN_HTTPS: 'false'
       PHPLDAPADMIN_LDAP_HOSTS: 'ldap'
       LDAP_HOSTNAME: '{{ container_addr_map.phpldapadmin.hostname }}'
-    link: '{{ openldap_container_name }}:ldap'
+    networks:
+      - name: '{{ docker_network_frontend }}'
     image: '{{ phpldapadmin_image_name }}:{{ docker_image_tag }}'
 
 # *****************************************************************************
diff --git a/roles/docker-svn/defaults/main.yml b/roles/docker-svn/defaults/main.yml
index 03d47174..bb416344 100644
--- a/roles/docker-svn/defaults/main.yml
+++ b/roles/docker-svn/defaults/main.yml
@@ -29,3 +29,6 @@ svn_dv_name: '{{ docker_name_prefix }}svn_DV'
 # restore directories to temporarly store data being restored into docker containers
 svn_docker_backup_dir: '{{ docker_backup_dir }}/SVN'
 host_svn_docker_restore_dir: '{{ bacula_dest }}{{ svn_docker_backup_dir }}'
+
+# Network names
+docker_network_frontend: 'frontend'
diff --git a/roles/docker-svn/tasks/main.yml b/roles/docker-svn/tasks/main.yml
index c7ae044e..c458e282 100644
--- a/roles/docker-svn/tasks/main.yml
+++ b/roles/docker-svn/tasks/main.yml
@@ -82,8 +82,8 @@
     restart_policy: always
     hostname: '{{ container_addr_map.svn.hostname }}'
     ports: '{{ container_port_map.svn.port_args }}'
-    link:
-      - '{{ openldap_container_name }}:ldap'
+    networks:
+      - name: '{{ docker_network_frontend }}'
     volumes:
       - '{{ openssl_dv_name }}:/etc/ssl/private:z'
       - '{{ svn_dv_name }}:/var/lib/svn:z'
